packages/Search/test/DTK_BoostRTreeHelpers.hpp

```diff
@@ -0,0 +1,115 @@
+#ifndef DTK_BOOST_RTREE_HELPERS_HPP
+#define DTK_BOOST_RTREE_HELPERS_HPP
+
+#include <DTK_DetailsBox.hpp>
+#include <DTK_DetailsPoint.hpp>
+#include <DTK_DetailsPredicate.hpp>
+
+#include "DTK_BoostGeometryAdapters.hpp"
+#include "DTK_BoostRangeAdapters.hpp"
+#include <boost/range/adaptors.hpp>
+
+namespace BoostRTreeHelpers
+{
+// NOTE: The balancing algorithm does not really matter since the tree is
+// created using packing algorithm and there is no later insertion of new
+// objects into the tree.  The values of the maximum and the minimum number
+// of elements could be adjusted in principle but I didn't bother setting
+// this type as a template argument.
+using Parameter = boost::geometry::index::linear<16>;
+
+template <typename Value, typename Index>
+struct PairMaker
+{
+    template <typename T>
+    inline std::pair<Value, Index> operator()( T const &v ) const
+    {
+        return std::make_pair( v.value(), v.index() );
+    }
+};
+
+template <typename Indexable>
+using RTree =
+    boost::geometry::index::rtree<std::pair<Indexable, int>, Parameter>;
+
+// NOTE: The template parameter Indexable is used as an alias here
```

Comment:
Why do you do that? Can I use a different `Indexable`?

Reply:
No you cannot.  I was trying to avoid writing `typename View::value_type` more than once :/